src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -244,6 +248,99 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
             }
         }
 
+        internal override T ReadAsPropertyNameCore(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+#if NETCOREAPP
+            bool success = TryParseEnumCore(ref reader, options, out T value);
+#else
+            bool success = TryParseEnumCore(reader.GetString(), options, out T value);
+#endif
+
+            if (!success)
+            {
+                ThrowHelper.ThrowJsonException();
+            }
+
+            return value;
+        }
+
+        internal override void WriteAsPropertyNameCore(Utf8JsonWriter writer, T value, JsonSerializerOptions options, bool isWritingExtensionDataProperty)
+        {
+            ulong key = ConvertToUInt64(value);
+            string original = value.ToString();
+
+            if (IsValidIdentifier(original))
+            {
+                if (options.DictionaryKeyPolicy != null)
+                {
+                    original = options.DictionaryKeyPolicy.ConvertName(original);
```

Comment:
Shouldn't this also use the same method for consistency in comma-separate values? This would need another test.
```suggestion
                    original = FormatUsingNamingPolicy(original, options.DictionaryKeyPolicy);
```

Reply:
👍 yay bug fix.